common/file_reader.h

```diff
@@ -410,6 +410,17 @@ class FileReader {
     itoa(v, value, 10);
     write_key_value(key, value);
   }
+  void write_key_value_float(const char* key, float f) {
+    int  a_ = (int)floorf(f);
+    int  b_ = ((int)floorf(f * 10)) % 10;
+    int  c_ = ((int)floorf(f * 100)) % 10;
+    char new_value[10];
```

Comment:
This needs to be bigger.
An int can take 10 bytes, then '.' and 2 decimals, plus a zero.
Just make it 16 bytes.

Reply:
I have made 16 bytes